tests/e2e/tests/pilot/pilot_test.go

```diff
@@ -152,16 +152,43 @@ func runRetriableTest(t *testing.T, cluster, testName string, retries int, f fun
 	})
 }
 
+type resource struct {
+	// Kind of the resource
+	Kind string
+	// Name of the resource
+	Name string
+}
+
 // deployableConfig is a collection of configs that are applied/deleted as a single unit.
 type deployableConfig struct {
-	Namespace  string
-	YamlFiles  []string
+	Namespace string
+	YamlFiles []string
+	// List of resources must be removed during deployableConfig setup, and restored
+	// during teardown. Typically, these are resources added by default installation and need to
+	// be modified for test. These resources are in the same namespace defined above.
+	Removes    []resource
```

Comment:
Are users of this structure required to populate the 'Removes' list? Maybe you want to point it out in the comments.